internal/test/framework/framework.go

```diff
@@ -149,12 +155,19 @@ func (f *TestFramework) Setup(vmCount int, credentials []*types.Credentials, ski
 			return fmt.Errorf("unable to instantiate Windows VM: %v", err)
 		}
 	}
-	if err := f.getKubeClient(); err != nil {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
```

Comment:
Have you confirmed that `config` can be reused across different clients?

Reply:
I've looked at each of the clients and I dont see anything that suggests that it can't be used across them.
Each client makes a copy of the config and then uses the copy and not the original.